### PR TITLE
Fix dependencies issues in the cmake machinery

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This should fix what @S-Dafarra and @isorrentino spotted in https://github.com/ami-iit/bipedal-locomotion-framework/issues/413

### CODE REVIEW

**Review by S-Dafarra — APPROVED** [review]
Thanks!

**Comment by traversaro** [comment]
It is possible that we did not spotted this as we are missing some dependency in https://github.com/ami-iit/bipedal-locomotion-framework/blob/252e9308b91f034bafc90388755336a447002801/.github/workflows/ci.yml#L487 ?

**Comment by S-Dafarra** [comment]
> It is possible that we did not spotted this as we are missing some dependency in
> 
> https://github.com/ami-iit/bipedal-locomotion-framework/blob/252e9308b91f034bafc90388755336a447002801/.github/workflows/ci.yml#L487
> 
> ?

Good point!

**Comment by GiulioRomualdi** [comment]
Lovely `vcpkg`

```
2021-09-16T13:52:49.0573896Z CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
2021-09-16T13:52:49.0579814Z     Command failed: msbuild C:/robotology/vcpkg/buildtrees/python3/x64-windows-rel/v3.9.6-ba0c8fc95d.clean/PCbuild/pcbuild.proj /p:Configuration=Release /p:IncludeExtensions=true /p:IncludeExternals=true /p:IncludeCTypes=true /p:IncludeSSL=true /p:IncludeTkinter=false /p:IncludeTests=false /p:ForceImportBeforeCppTargets=C:/robotology/vcpkg/buildtrees/python3/src/v3.9.6-ba0c8fc95d.clean/PCbuild/python_vcpkg.props /p:WindowsTargetPlatformVersion=10.0.22000.0 /p:DefaultWindowsSDKVersion=10.0.22000.0 /p:IncludeUwp=false /p:_VcpkgPythonLinkage=DynamicLibrary /t:Rebuild /p:Platform=x64 /p:PlatformToolset=v142 /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion=10.0.22000.0 /p:VcpkgTriplet=x64-windows /p:VcpkgInstalledDir=C:/robotology/vcpkg/installed /p:VcpkgManifestInstall=false /m
2021-09-16T13:52:49.0584837Z     Working Directory: C:/robotology/vcpkg/buildtrees/python3/x64-windows-rel/v3.9.6-ba0c8fc95d.clean
2021-09-16T13:52:49.0585620Z     Error code: 1
2021-09-16T13:52:49.0586125Z     See logs for more information:
2021-09-16T13:52:49.0586958Z       C:\robotology\vcpkg\buildtrees\python3\build-x64-windows-rel-out.log
2021-09-16T13:52:49.0587685Z 
2021-09-16T13:52:49.0588162Z Call Stack (most recent call first):
2021-09-16T13:52:49.0588932Z   scripts/cmake/vcpkg_install_msbuild.cmake:169 (vcpkg_execute_required_process)
2021-09-16T13:52:49.0589799Z   ports/python3/portfile.cmake:115 (vcpkg_install_msbuild)
2021-09-16T13:52:49.0591172Z   scripts/ports.cmake:141 (include)
2021-09-16T13:52:49.0592210Z 
2021-09-16T13:52:49.0592530Z 
2021-09-16T13:52:49.0593353Z Error: Building package python3:x64-windows failed with: BUILD_FAILED
2021-09-16T13:52:49.0595081Z Please ensure you're using the latest portfiles with `.\vcpkg update`, then
2021-09-16T13:52:49.0596252Z submit an issue at https://github.com/Microsoft/vcpkg/issues including:
2021-09-16T13:52:49.0597275Z   Package: python3:x64-windows
2021-09-16T13:52:49.0598123Z   Vcpkg version: 2021-07-21-244f69e78da4f97034328b4d290db262d748d4c7
2021-09-16T13:52:49.0598842Z 
2021-09-16T13:52:49.0599691Z Additionally, attach any relevant sections from the log files above.
2021-09-16T13:52:49.3784304Z ##[error]Process completed with exit code 1.
```

**Comment by GiulioRomualdi** [comment]
@traversaro, in https://github.com/ami-iit/bipedal-locomotion-framework/pull/414/commits/3e1e1056ae69fa9f9084afef11449d5737af1898 I updated the CI test compilation

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -484,7 +484,7 @@ jobs:
       shell: bash
       run: |
         cd build
-        for missing_dep in YARP Qhull casadi cppad manif Python3 pybind11 pytest matioCpp LieGroupControllers; do
+        for missing_dep in YARP Qhull casadi cppad manif Python3 pybind11 pytest matioCpp LieGroupControllers nlohmann_json; do
```
Comment (traversaro) [inline .github/workflows/ci.yml:487]: `spdlog` is also missing?

Comment (GiulioRomualdi) [reply .github/workflows/ci.yml:487]: `spdlog` is a mandatory dependency :)

**Pull request merged**